system_test/arrow_ingress_fuzz.py

```
@@ -0,0 +1,927 @@
+def _iso_to_us(s: str) -> int:
+    """ISO datetime string → microseconds since epoch (handles ns suffix)."""
+    s = s.rstrip("Z")
+    if "." in s:
+        head, frac = s.split(".", 1)
+        if "T" not in head:
+            head = head.replace(" ", "T")
+        frac = frac.ljust(6, "0")
+        us = int(frac[:6])
+        ns_tail = frac[6:]
+        if ns_tail and any(c != "0" for c in ns_tail):
+            us += int(round(int(ns_tail.ljust(3, "0")[:3]) / 1000.0))
```

Ruff (0.15.14)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)